File: WEEK_3_Python/Q5.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def nth_from_end(head, n):
    fast = slow = head
    for _ in range(n):
        if not fast: return None
        fast = fast.next
    while fast:
        slow = slow.next
        fast = fast.next
    return slow.data if slow else None

# Example
head = Node(1); head.next = Node(2); head.next.next = Node(3); head.next.next.next = Node(4)
print("2nd from end:", nth_from_end(head, 2))   

File: WEEK_3_C/Q5.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

int main(){
    struct Node *head=NULL,*t;
    for(int i=1;i<=4;i++){ t=(struct Node*)malloc(sizeof(struct Node)); t->data=i*10; t->next=head; head=t; }
    int key=20,found=0;
    for(t=head;t;t=t->next) if(t->data==key){ found=1; break; }
    if(found) printf("Found\n"); else printf("Not Found\n");
    return 0;
}